src/blog/2024/03/flowfuse-on-siemens-iot2050.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Run FlowFuse on Siemens IoT2050
+subtitle: How to install FlowFuse Device Agent on Siemens IoT2050
+description: In this guide, we will discuss how to install FlowFuse Device agent on Siemens IoT2050.
+date: 2024-03-04
+authors: ["grey-dziuba"]
+image: 
+tags:
+    - posts
+    - flowfuse
+---
+
+Siemens [announced](https://press.siemens.com/global/en/pressrelease/new-siemens-gateway-between-cloud-company-it-and-production) the IoT2000 series in March of 2020.  With this tool many have been using it to function as a gateway between their plant operations and cloud infrastructure.  Onboard it came with Node-RED pre installed.  In this guide we will show you how to install the FlowFuse device agent.  
+
+<!--more-->
+
+<div style="background-color: #fff4b9; border:1px solid #ffc400; color: #a27110; padding: 12px; border-radius: 6px; font-style: italic;">Warning: Later in the guide we will be updating Node.js. This will break mraa library. This will prevent communication to the I/O of the device.</div>
+
+## Prerequisites 
+
+In this guide we will be working with the IoT2050 Advanced, *6ES7 647-0BA00-1YA2*. The device has been [upgraded](https://support.industry.siemens.com/cs/attachments/109741799/IOT2050_How_To_Firmware_Update_V1.3.pdf) to the latest firmware at the time of writing this article of v1.3.1.  We will be leveraging the IOT2050_Example_Image_V1.3.1.zip image which is a debian base OS.  To complete this guide, knowledge of linux based cli is necessary.  Documentation to complete these requirements can be found [here](https://support.industry.siemens.com/cs/document/109741799/downloads-for-simatic-iot20x0?dti=0&lc=en-GB).  Our boot order is the following:
+
+## Guide
+
+1. First we need to run the standard updates.
+
+```shell
+apt-get update
+apt-get upgrade
+```
+
+2. If you need to migrate your existing Node-RED follow these [instruction](https://flowfuse.com/docs/migration) to backup your existing progress.  From there we will need to remove the existing service that autostart Node-RED by running the following command and reboot:
+
+```shell
+systemctl disable node-red.service
+reboot -h now
+```
+
+3. Confirm that your Node-RED instance is no longer running.
+
+```shell
+systemctl status node-red
+```
+
+4. Now it is time to upgrade your Node.js version.  To check the version before we get started run ```node -v```.  Then install a tool that will allow you to change your versions of Node.js with the following command.
+
+<div style="background-color: #fff4b9; border:1px solid #ffc400; color: #a27110; padding: 12px; border-radius: 6px; font-style: italic;">Warning: updating Node.js will break mraa library. This will prevent communication to the I/O of the device.</div>
```

Comment:
Can we link to an external discussion to track to see if it gets updated to work?

Reply:
```suggestion
<div style="background-color: #fff4b9; border:1px solid #ffc400; color: #a27110; padding: 12px; border-radius: 6px; font-style: italic;">Warning: updating Node.js will break mraa library. This will prevent communication to the I/O of the device. Details can be found <a href="https://support.industry.siemens.com/forum/WW/en/posts/iot2050-node-js-versions/297170">here</a>.</div>
```